specification/_types/Knn.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Field } from '@_types/common'
+import { long, double, float } from '@_types/Numeric'
+import { QueryContainer } from './query_dsl/abstractions'
+
+export type KnnQueryVector = double[]
+
+export interface KnnQuery {
+  /** The name of the vector field to search against */
+  field: Field
+  /** The query vector */
+  query_vector: KnnQueryVector
```

Comment:
We could just use `double[]` here, as KnnQueryVector doesn't seem to have any special semantics besides being a vector of numbers.